tequilapi/http_middlewares.go

```diff
@@ -7,12 +7,32 @@ type corsHandler struct {
 }
 
 func (wrapper corsHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
-	resp.Header().Set("Access-Control-Allow-Origin", "*")
-	resp.Header().Set("Access-Control-Allow-Methods", "POST, GET, OPTIONS, PUT, DELETE")
+	if isPreflightCorsRequest(req) {
+		applyPreflightCorsResponse(resp)
+		return
+	}
+
+	allowAllCorsActions(resp)
 	wrapper.originalHandler.ServeHTTP(resp, req)
 }
 
 //ApplyCors wraps original handler by adding cors headers to response BEFORE original ServeHTTP method is called
 func ApplyCors(original http.Handler) http.Handler {
 	return corsHandler{original}
 }
+
+func allowAllCorsActions(resp http.ResponseWriter) {
+	resp.Header().Set("Access-Control-Allow-Origin", "*")
+	resp.Header().Set("Access-Control-Allow-Methods", "POST, GET, OPTIONS, PUT, DELETE")
+}
+
+func isPreflightCorsRequest(req *http.Request) bool {
+	isOptionsMethod := req.Method == http.MethodOptions
+	containsAccessControlRequestMethod := req.Header.Get("Access-Control-Request-Method") != ""
+	containsAccessControlRequestHeader := req.Header.Get("Access-Control-Request-Headers") != ""
+	return isOptionsMethod && containsAccessControlRequestHeader && containsAccessControlRequestMethod
+}
+
+func applyPreflightCorsResponse(resp http.ResponseWriter) {
```

Comment:
Why conditional check is done?
As i understand, later both `applyPreflightCorsResponse()` and `allowAllCorsActions()` does the same

Reply:
Actually there are two cases:
1. If it's preflight request - add headers and return EMPTY response
2. Any other case - simply add cors headers.
I wanted to show intent that it's separate case - return empty response with headers. Maybe renaming function to generatePreflightCorsResponse will express intetion better.

Reply:
What about this?
```go
allowAllCorsActions(resp)
if isPreflightCorsRequest(req) {
  return
}
wrapper.originalHandler.ServeHTTP(resp, req)
```
This shows that same headers are applied always and that empty response is returned in case it's preflight request.